### PR TITLE
feat(DDI-1101): Added slots for leading and trailing contents

### PR DESCRIPTION
**Pull request by Radha-Alberta**
Added two new properties leadingContent and trailingContent in the input component.
Added warning message to show when prefix/suffix are used.
Updated tests accordingly.

### CODE REVIEW

File: libs/docs/src/components/common/Input.stories.mdx [hunk 1]
```diff
@@ -166,6 +166,34 @@ import { Meta, Story } from "@storybook/addon-docs";
...
+    defaultValue=""
+    description="Sets the content to the right of the input field"
+  />
+  <Prop
+    name="trailingContent"
+    type="ReactNode"
+    lang="react"
+    defaultValue=""
+    description="Sets the content to the right of the input field"
+  />
```
[the remark this answers is not in the document]
Comment (ArakTaiRoth) [reply libs/docs/src/components/common/Input.stories.mdx:196]: Oh, and remove the examples using prefix and suffix from the Storybook as well

File: libs/react-components/src/lib/input/input.tsx [hunk 2]
```diff
@@ -95,6 +95,8 @@ export interface InputProps extends BaseProps {
   min?: number | string;
   max?: number | string;
   step?: number;
+  leadingContent?: React.ReactNode;
+  trailingContent?: React.ReactNode;
```
Comment (chrisolsen) [inline libs/react-components/src/lib/input/input.tsx:99]: These should be moved into the BaseProps interface

File: libs/react-components/src/lib/input/input.spec.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { render } from "@testing-library/react";
 import { fireEvent, screen } from "@testing-library/dom";
 import { GoAInputText, InputProps } from "./input";
 import { GoAIconType } from "../icon/icon";
+import React, { useState } from "react";
```
Comment (chrisolsen) [inline libs/react-components/src/lib/input/input.spec.tsx:6]: I don't think you are using the `useState`

File: libs/docs/src/components/common/Input.stories.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisolsen) [reply libs/docs/src/components/common/Input.stories.mdx:196]: Should the be removed or marked as deprecated?

File: libs/docs/src/components/common/Input.stories.mdx [hunk 4]
```diff
@@ -987,6 +1015,55 @@ export const AutocapitalizeTemplate = () => {
   </Tab>
 </Tabs>
 
+#### LeadingContent and TrailingContent
```
Comment (chrisolsen) [inline libs/docs/src/components/common/Input.stories.mdx:1018]: I think there should be an additional example that includes adding a one of the web components to the slot so it is clear that it isn't text only.

**Comment by chrisolsen** [comment]
leadingIcon and trailingIcon also need to be included what is being moved to the new slots

**Comment by chrisolsen** [comment]
You will also need to fix your commit message. Your DDI-XXX has a trailing underscore instead of closig paren

File: libs/react-components/src/lib/input/input.tsx (hunk 2 above)
Comment (Radha-Alberta) [reply libs/react-components/src/lib/input/input.tsx:99]: Done.

File: libs/react-components/src/lib/input/input.spec.tsx (hunk 3 above)
Comment (Radha-Alberta) [reply libs/react-components/src/lib/input/input.spec.tsx:6]: Yes, removed useState from import

**Comment by Radha-Alberta** [comment]
> You will also need to fix your commit message. Your DDI-XXX has a trailing underscore instead of closig paren

Fixed

File: libs/docs/src/components/common/Input.stories.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Radha-Alberta) [reply libs/docs/src/components/common/Input.stories.mdx:196]: Done. Marked them as deprecated.

File: libs/docs/src/components/common/Input.stories.mdx [hunk 5]
```diff
@@ -147,12 +147,12 @@ import { Meta, Story } from "@storybook/addon-docs";
   <Prop
     name="prefix"
     type="string"
-    description="Text that comes before the input"
+    description="Text that comes before the input. **Deprecated**"
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Input.stories.mdx:150]: For some reason this doesn't make the `Deprecated` bold in the storybook. This should definitely be made bold, if possible.

File: libs/docs/src/components/common/Input.stories.mdx [hunk 6]
```diff
@@ -987,6 +1019,55 @@ export const AutocapitalizeTemplate = () => {
...
+#### LeadingContent and TrailingContent
+
+<Story name="LeadingContent and TrailingContent">
+  <>
+    <h5>Leading Content</h5>
+    <GoAInput name="input" value="" leadingContent="$" onChange={noop} />
+    <h5>Trailing Content</h5>
+    <GoAInput name="input" value="" trailingContent="items" onChange={noop} />
+    <h5>Leading and Trailing Content</h5>
+    <GoAInput
```
Comment (ArakTaiRoth) [inline libs/docs/src/components/common/Input.stories.mdx:1031]: Can we change one or more of these examples to showcase how different things other than just strings can be used in here? One example definitely needs to be a div with its own styled width.

Comment (Radha-Alberta) [reply libs/docs/src/components/common/Input.stories.mdx:1045]: Added examples

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**